src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Rfc2898DeriveBytes.cs

```diff
@@ -84,34 +84,36 @@ public Rfc2898DeriveBytes(string password, int saltSize, int iterations, HashAlg
             HashAlgorithm = hashAlgorithm;
             _hmac = OpenHmac(passwordBytes);
             CryptographicOperations.ZeroMemory(passwordBytes);
-            // _blockSize is in bytes, HashSize is in bits.
-            _blockSize = _hmac.HashSize >> 3;
+            _blockSize = _hmac.HashLengthInBytes;
 
             Initialize();
         }
 
-        internal Rfc2898DeriveBytes(byte[] password, byte[] salt, int iterations, HashAlgorithmName hashAlgorithm, bool clearPassword)
+        internal Rfc2898DeriveBytes(byte[] password, byte[] salt, int iterations, HashAlgorithmName hashAlgorithm, bool clearPassword) :
+            this(
+                new ReadOnlySpan<byte>(password ?? throw new NullReferenceException()), // This "should" be ArgumentNullException but for compat, we throw NullReferenceException.
```

Comment:
@bartonjs, do we need to maintain this level of "compat" moving forward?  We've fixed lots of null reference exceptions, and we're well past trying for any kind of bug-for-bug compatibility.

Reply:
Nah, probably not.  It's just an artifact of seeding from netfx and having unified netfx/corefx tests.  And now it just moved.